scripts/templates/installer.sh

```diff
@@ -956,45 +967,60 @@ mount_all() {
         mount --move /system /system_root
       fi
       if [ $? != 0 ]; then
-        umount /system
-        umount -l /system 2>/dev/null
-        if [ "$dynamic_partitions" = "true" ]; then
-          test -e /dev/block/mapper/system || local slot=$(getprop ro.boot.slot_suffix 2>/dev/null)
-          mount -o ro -t auto /dev/block/mapper/system$slot /system_root
-          mount -o ro -t auto /dev/block/mapper/vendor$slot /vendor 2>/dev/null
+        (umount /system
+        umount -l /system) 2>/dev/null
+        if [ -d /dev/block/mapper ]; then
+          test -e /dev/block/mapper/system || local slot=$(find_slot)
+          mount -o ro -t auto /dev/block/mapper/vendor$slot /vendor
           mount -o ro -t auto /dev/block/mapper/product$slot /product 2>/dev/null
+          mount -o ro -t auto /dev/block/mapper/system$slot /system_root
         else
-          test -e /dev/block/bootdevice/by-name/system || local slot=$(getprop ro.boot.slot_suffix 2>/dev/null)
+          test -e /dev/block/bootdevice/by-name/system || local slot=$(find_slot)
+          (mount -o ro -t auto /dev/block/bootdevice/by-name/vendor$slot /vendor
+          mount -o ro -t auto /dev/block/bootdevice/by-name/product$slot /product
+          mount -o ro -t auto /dev/block/bootdevice/by-name/persist$slot /persist) 2>/dev/null
           mount -o ro -t auto /dev/block/bootdevice/by-name/system$slot /system_root
         fi
       fi
     ;;
   esac
   if is_mounted /system_root; then
+    mount_apex
     if [ -f /system_root/build.prop ]; then
       mount -o bind /system_root /system
     else
       mount -o bind /system_root/system /system
     fi
   fi
-  mount_apex
 }
 
 umount_all() {
-  (umount_apex
-  umount /system
+  (umount /system
   umount -l /system
   if [ -e /system_root ]; then
     umount /system_root
     umount -l /system_root
   fi
-  for p in "/cache" "/persist" "/vendor" "/product"; do
-    umount $p
-    umount -l $p
-  done
+  umount /mnt/system
+  umount -l /mnt/system
+  umount_apex
+  umount /vendor
+  umount -l /vendor
+  umount /mnt/vendor
+  umount -l /mnt/vendor
+  umount /product
+  umount -l /product
+  umount /mnt/product
+  umount -l /mnt/product
+  umount /persist
+  umount -l /persist
```

Comment:
why not umount `/mnt/persist` too?

And these entries after unmounting after apex can still be a for-loop, right?

Reply:
Didn't know /mnt/persist was a thing, and arguably you could just mount it twice for your purposes (.gapps-config).

Also, yes, they could be a for-loop, but I chose to leave them as individual commands so it's closer to my version in AnyKernel3 for the same upkeep reasons you bring up below. 👍

Reply:
I don't know if it is a thing. But it just occurred it is the only one that has no /mnt entry

Keeping it close to your AK3 is good :-)

Reply:
Just double-checked on my OnePlus 8 Pro, and on Lineage Recovery the fstab has `
/dev/block/bootdevice/by-name/persist /mnt/vendor/persist` and it's not mounted by default, so I think it should be find to leave as-is, since that won't likely ever get mounted.

Reply:
Updated AK3 with the additional Open GApps partitions, and added the for-loop. Should be good here now.